src/Sema.zig

```diff
@@ -22607,7 +22625,9 @@ fn beginComptimePtrMutation(
             }
         },
         .opt_payload_ptr => {
-            const opt_ptr = ptr_val.castTag(.opt_payload_ptr).?.data;
+            const opt_ptr = if (ptr_val.castTag(.opt_payload_ptr)) |some| some.data else {
```

Comment:
Is the else here `unreachable`? Hopefully I'm not missing something

Reply:
I got an `attempt to use null value` here when running the added test.